internal/k8s/deployments_test.go

```diff
@@ -557,3 +566,238 @@ func TestComputeConnectionSpecHash(t *testing.T) {
+
+	tests := map[string]struct {
+		connection      temporaliov1alpha1.TemporalConnectionSpec
+		expectedAddress string
+		expectTLS       bool
+	}{
+		"without TLS": {
+			connection: temporaliov1alpha1.TemporalConnectionSpec{
+				HostPort: "test.temporal.example:9999",
+			},
+			expectedAddress: "test.temporal.example:9999",
+			expectTLS:       false,
+		},
+		"with TLS": {
+			connection: temporaliov1alpha1.TemporalConnectionSpec{
+				HostPort:        "mtls.temporal.example:8888",
+				MutualTLSSecret: "test-tls-secret",
+			},
+			expectedAddress: "mtls.temporal.example:8888",
+			expectTLS:       true,
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			spec := &temporaliov1alpha1.TemporalWorkerDeploymentSpec{
+				Template: corev1.PodTemplateSpec{
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:  "worker",
+								Image: "temporal/worker:latest",
+							},
+						},
+					},
+				},
+				WorkerOptions: temporaliov1alpha1.WorkerOptions{
+					TemporalNamespace: "test-namespace",
+				},
+			}
+
+			deployment := k8s.NewDeploymentWithOwnerRef(
+				&metav1.TypeMeta{},
+				&metav1.ObjectMeta{Name: "test", Namespace: "default"},
+				spec,
+				"test-deployment",
+				"test-build-id",
+				tt.connection,
+			)
+
+			// Extract environment variables from the deployment
+			container := deployment.Spec.Template.Spec.Containers[0]
+
+			// Set environment variables using t.Setenv() to simulate the runtime environment
+			for _, env := range container.Env {
+				t.Setenv(env.Name, env.Value)
+			}
+
+			if tt.expectTLS {
+				// Create temporary test certificate files
+				tempDir := t.TempDir()
+				certPath := filepath.Join(tempDir, "client.pem")
+				keyPath := filepath.Join(tempDir, "client.key")
+
+				// Generate a self-signed certificate for testing
+				privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+				require.NoError(t, err)
+
+				template := x509.Certificate{
+					SerialNumber: big.NewInt(1),
+					Subject: pkix.Name{
+						CommonName: "test",
+					},
+					NotBefore:   time.Now(),
+					NotAfter:    time.Now().Add(365 * 24 * time.Hour),
+					KeyUsage:    x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+					ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
+				}
+
+				certDER, err := x509.CreateCertificate(rand.Reader, &template, &template, &privateKey.PublicKey, privateKey)
+				require.NoError(t, err)
+
+				// Write certificate file
+				certPEM := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: certDER})
+				err = os.WriteFile(certPath, certPEM, 0644)
+				require.NoError(t, err)
+
+				// Write private key file
+				keyDER, err := x509.MarshalPKCS8PrivateKey(privateKey)
+				require.NoError(t, err)
+				keyPEM := pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: keyDER})
+				err = os.WriteFile(keyPath, keyPEM, 0644)
+				require.NoError(t, err)
```

Comment:
Extract all of the cert creation into a test helper func (should call `t.Helper()`)